recipes/flint/all/conanfile.py

```diff
@@ -0,0 +1,84 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+class FlintConan(ConanFile):
+    name = "flint"
+    description = "FLINT (Fast Library for Number Theory)"
+    license = "LGPL-2.1-or-later"
+    topics = ("conan", "math", "numerical")
+    homepage = "https://www.flintlib.org"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("gmp/6.2.1")
+        self.requires("mpfr/4.1.0")
+        if self.settings.compiler == "Visual Studio":
+            self.requires("pthreads4w/3.0.0")
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
```

Comment:
```suggestion
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
```

Reply:
Is it recommended for all new recipes, even with a single version?

Reply:
There's no harm and it's future proof without code bloat.